src/backend/executor/nodeShareInputScan.c

```diff
@@ -925,9 +923,12 @@ writer_wait_for_acks(ShareInput_Lk_Context *pctxt, int share_id, int xslice)
 			int save_errno = errno;
 			elog(LOG, "SISC WRITER (shareid=%d, slice=%d): notify still wait for an answer, errno %d",
 					share_id, currentSliceId, save_errno);
-			/*if error(except EINTR) happens in select, we just return to avoid endless loop*/
-			if(errno != EINTR){
-				return;
+			if(save_errno == EBADF)
+			{
+				/* The file description is invalid, maybe this FD has been already closed by writer in some cases
+				 * we need to break here to avoid endless loop and continue to run CHECK_FOR_INTERRUPTS.
+				 */
+				break;
```

Comment:
select() ERRORS :
EBADF  -- break;
EINTR   -- loop
EINVAL -- programming error, should not occurs

on Linux:
ENOMEM -- loop, wait for runaway to choose one transaction to rollback, or OS choose one process to kill

On macos:
EAGAIN -- loop

EAGAIN -- loop

So we just process the EBADF only, others are loop again or impossible to occurs. Thanks.